content/apps/continuous-deployment.md

```diff
@@ -0,0 +1,111 @@
+---
+menu:
+  main:
+    parent: apps
+title: Continuous Deployment
+weight: 10
+---
+
+Setting up continuous deployment allows you to automatically upload your
+changes to your desired enivronment.
+
+The first thing you need to do is setup a "deployer" user in your organization and give it permission to deploy to the desired space:
+
+```
+cf create-user ORGNAME_deployer RANDOM-PASSWORD
+cf set-space-role ORGNAME_deployer ORG SPACE SpaceDeveloper
+```
+
+***
+
+Depending on your CI system the setup is going to be a bit different.
+
+### Travis
+
+Per [http://docs.travis-ci.com/user/deployment/cloudfoundry/](http://docs.travis-ci.com/user/deployment/cloudfoundry/) you need to add this section to your `.travis.yml` file:
+
+```
+deploy:
+  edge: true
+  provider: cloudfoundry
+  username: DEPLOYER_USER
+  password:
+  api: https://api.cf.18f.us
+  organization: ORG
+  space: SPACE
+```
+
+Replace `DEPLOYER_USER`, `ORG`, and `SPACE` accordingly and run `travis encrypt --add deploy.password --skip-version-check` to enter your password.
+
+**THIS SOLUTION DOESN'T CURRENTLY WORK SINCE TRAVIS DOES NOT SUPPORT `--skip-ssl-validation` for CF**
+
+***
+
+### CircleCI
+
+Add this items to your `circle.yml` file:
+
+```
+dependencies:
+  pre:
+    - wget http://go-cli.s3-website-us-east-1.amazonaws.com/releases/latest/cf-cli_amd64.deb -qO temp.deb
+    - sudo dpkg -i temp.deb
+    - cf -v
+
+test:
+  post:
+    - cf login -a https://api.cf.18f.us -u DEPLOYER_USER -p $CF_PASS -o ORG -s SPACE --skip-ssl-validation
+
+deployment:
+  production:
+    branch: master
+    commands:
+      - cf push
```

Comment:
Since there is a requirement for manifest.yml it kind of assumes it has to push the whole thing but it is a good point... I am adding a note to "you might want to specify which app to push"...